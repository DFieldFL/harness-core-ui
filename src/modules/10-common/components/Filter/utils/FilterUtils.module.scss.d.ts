/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly noStyleUl: string
  readonly summaryItem: string
}
export default styles
