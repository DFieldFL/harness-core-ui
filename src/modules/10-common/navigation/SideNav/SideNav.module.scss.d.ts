/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly main: string
  readonly titleContainer: string
  readonly title: string
  readonly iconContainer: string
  readonly icon: string
  readonly link: string
  readonly selected: string
  readonly text: string
}
export default styles
