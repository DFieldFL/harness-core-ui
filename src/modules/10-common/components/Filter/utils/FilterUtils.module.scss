.noStyleUl {
  list-style-type: none;
  padding: var(--spacing-small);
  margin: 0;
  background-color: var(--black);
  color: var(--white);
}

.summaryItem {
  padding: var(--spacing-xsmall);
  font-size: small;
}
