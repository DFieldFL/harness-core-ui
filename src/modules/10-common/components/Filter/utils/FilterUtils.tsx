import { isEmpty, isUndefined, omitBy } from 'lodash-es'
import React from 'react'
import type { FilterDTO } from 'services/cd-ng'

import css from './FilterUtils.module.scss'

type supportedTypes = string | number | boolean | object

const tagSeparator = ' : '

export const renderItemByType = (data: supportedTypes | Array<supportedTypes> | object): string => {
  if (Array.isArray(data)) {
    return data.join(', ')
  } else if (typeof data === 'object') {
    return Object.entries(data as object)
      .map(([key, value]) => {
        return key.toString().concat(value ? tagSeparator.concat(value.toString()) : '')
      })
      .join(', ')
  } else if (typeof data === 'number') {
    return data.toString()
  } else if (typeof data === 'boolean') {
    return data ? 'true' : 'false'
  }
  return typeof data === 'string' ? data : ''
}

export const getFilterSummary = (fieldToLabelMapping: Map<string, string>, fields: object): JSX.Element => {
  return (
    <ol className={css.noStyleUl}>
      {Object.entries(fields as object).map(([key, value]) => {
        if (fieldToLabelMapping.has(key)) {
          return (
            <li key={key} className={css.summaryItem}>
              <span>{fieldToLabelMapping.get(key)} : </span>
              <span style={{ fontWeight: 'bold' }}>{renderItemByType(value)}</span>
            </li>
          )
        }
      })}
    </ol>
  )
}

export const removeNullAndEmpty = (object: Record<string, any>) => omitBy(omitBy(object, isUndefined), isEmpty)
export const isObjectEmpty = (arg: Record<string, any>): boolean => isEmpty(omitBy(arg, isEmpty))
export const UNSAVED_FILTER = 'Unsaved Filter'
export const MAX_FILTER_NAME_LENGTH = 25

export const flattenObject = (obj: Record<string, any>) => {
  const flattened: {
    [key: string]: string
  } = {}
  Object.keys(obj).forEach((key: string) => {
    if (Object.prototype.toString.call(obj[key]) === '[object Object]' && obj[key] !== null) {
      Object.assign(flattened, flattenObject(obj[key]))
    } else {
      flattened[key as string] = obj[key]
    }
  })
  return flattened
}

export const getFilterSize = (obj: Record<string, any>): number => {
  return Object.keys(flattenObject(obj)).length
}

export const getFilterByIdentifier = (filters: FilterDTO[], identifier: string): FilterDTO | undefined =>
  filters?.find((filter: FilterDTO) => filter.identifier?.toLowerCase() === identifier.toLowerCase())
