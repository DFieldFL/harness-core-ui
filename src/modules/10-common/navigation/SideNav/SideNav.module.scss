.main {
  height: 100%;
  background: var(--theme-light-nav-panel-bg);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: relative;

  .bottomContainer {
    position: relative;

    .titleContainer {
      margin-left: var(--spacing-medium);
      margin-bottom: var(--spacing-medium);

      .title {
        font-size: 18px !important;
        letter-spacing: 1px;
      }
    }

    .iconContainer {
      position: absolute;
      bottom: 0;
      right: 0;
      overflow: hidden;
      height: 350px;
      width: 192px;

      .icon {
        position: absolute;
        left: -50px;
        bottom: -80px;
        opacity: 0.08;
      }
    }
  }
}

.link {
  display: block;
  margin-left: var(--spacing-medium);
  padding: var(--spacing-small) var(--spacing-medium);
  opacity: 0.8;

  &:hover {
    text-decoration: none;
    opacity: 1;
    border-top-left-radius: 2px;
    border-bottom-left-radius: 2px;
    background-color: rgba(2, 120, 213, 0.5);
  }

  &.selected {
    background-color: rgba(2, 120, 213, 0.8);
    border-top-left-radius: 2px;
    border-bottom-left-radius: 2px;
    opacity: 1;
  }

  .text {
    color: var(--white) !important;
    font-size: 13px !important;
  }
}
