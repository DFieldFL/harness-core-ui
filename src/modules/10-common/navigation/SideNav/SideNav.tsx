import React, { ReactElement } from 'react'
import cx from 'classnames'
import { NavLink as Link, NavLinkProps } from 'react-router-dom'
import { Text, Layout, Color, IconName, Icon } from '@wings-software/uicore'
import css from './SideNav.module.scss'

export interface SideNavProps {
  subtitle?: string
  title?: string
  icon?: IconName
}

export default function SideNav(props: React.PropsWithChildren<SideNavProps>): ReactElement {
  return (
    <div className={css.main}>
      <div>{props.children}</div>
      <div className={css.titleContainer}>
        {props.icon ? (
          <div className={css.iconContainer}>
            <Icon className={css.icon} name={props.icon} size={350} />
          </div>
        ) : null}
        <Layout.Vertical>
          {props.subtitle ? (
            <Text font={{ size: 'small' }} color={Color.GREY_400}>
              {props.subtitle}
            </Text>
          ) : null}
          {props.title ? (
            <Text color={Color.WHITE} className={css.title}>
              {props.title}
            </Text>
          ) : null}
        </Layout.Vertical>
      </div>
    </div>
  )
}

interface SidebarLinkProps extends NavLinkProps {
  label: string
  icon?: IconName
  className?: string
}

export const SidebarLink: React.FC<SidebarLinkProps> = ({ label, icon, className, ...others }) => (
  <Link className={cx(css.link, className)} activeClassName={css.selected} {...others}>
    <Text icon={icon} className={css.text}>
      {label}
    </Text>
  </Link>
)
