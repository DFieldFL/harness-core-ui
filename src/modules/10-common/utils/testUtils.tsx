import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uikit'
import qs from 'qs'

import strings from 'strings/strings.en.yaml'
import { AppStoreProvider, useAppStoreWriter } from 'framework/hooks/useAppStore'
import type { AppStore } from 'framework/types/AppStore'
import { AUTH_ROUTE_PATH_PREFIX, StringsContext } from 'framework/exports'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): Element | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): Element | null => document.querySelector('.bp3-popover-content')

interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, any>
  defaultAppStoreValues?: Partial<AppStore>
}

export const prependAccountPath = (path: string): string => AUTH_ROUTE_PATH_PREFIX + path

const AppStoreInitializer: React.FC<{ defaultAppStoreValues: TestWrapperProps['defaultAppStoreValues'] }> = ({
  children,
  defaultAppStoreValues
}) => {
  const updateAppStore = useAppStoreWriter()
  updateAppStore(defaultAppStoreValues || {})
  return <> {children} </>
}

export const NotFound = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  const { path = '/', pathParams = {}, defaultAppStoreValues, queryParams = {} } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  const history = createMemoryHistory({ initialEntries: [routePath] })

  return (
    <AppStoreProvider>
      <StringsContext.Provider value={strings}>
        <AppStoreInitializer defaultAppStoreValues={defaultAppStoreValues}>
          <Router history={history}>
            <ModalProvider>
              <RestfulProvider base="/">
                <Switch>
                  <Route exact path={path}>
                    {props.children}
                  </Route>
                  <Route>
                    <NotFound />
                  </Route>
                </Switch>
              </RestfulProvider>
            </ModalProvider>
          </Router>
        </AppStoreInitializer>
      </StringsContext.Provider>
    </AppStoreProvider>
  )
}
