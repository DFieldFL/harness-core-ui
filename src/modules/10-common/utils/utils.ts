import type { IconName } from '@wings-software/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}
