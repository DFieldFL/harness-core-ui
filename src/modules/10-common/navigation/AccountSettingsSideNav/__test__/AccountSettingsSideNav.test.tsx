import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import SideNav from '../AccountSettingsSideNav'

describe('Resources Page', () => {
  test('render', () => {
    const { container } = render(
      <TestWrapper path="/account/:accountId/admin/resources/" pathParams={{ accountId: 'dummy' }}>
        <SideNav />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
