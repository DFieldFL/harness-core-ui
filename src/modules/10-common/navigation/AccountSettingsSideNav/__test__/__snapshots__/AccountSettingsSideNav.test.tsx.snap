// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources Page render 1`] = `
<div>
  <div
    class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
  >
    <a
      class="link"
      href="/account/dummy/admin"
    >
      <p
        class="StyledProps--font StyledProps--main text"
      >
        Overview
      </p>
    </a>
    <a
      aria-current="page"
      class="link selected"
      href="/account/dummy/admin/resources"
    >
      <p
        class="StyledProps--font StyledProps--main text"
      >
        Resources
      </p>
    </a>
    <a
      class="link"
      href="/account/dummy/admin/access-control"
    >
      <p
        class="StyledProps--font StyledProps--main text"
      >
        Access Control
      </p>
    </a>
    <a
      class="link"
      href="/account/dummy/admin/organizations"
    >
      <p
        class="StyledProps--font StyledProps--main text"
      >
        Organizations
      </p>
    </a>
  </div>
</div>
`;
